### PR TITLE
Use the Spring Initializr

### PR DESCRIPTION
**Pull request by Buzzardo**
I rewrote the document and the code (mostly changing package names) to use the Spring Initializr. This particular example involved modifying both the Maven and the Gradle build files, which added quite a bit to the document.

### CODE REVIEW

File: complete/application/build.gradle [hunk 1]
```diff
@@ -1,30 +1,20 @@
...
-repositories { mavenCentral() }
 
 dependencies {
-    compile('org.springframework.boot:spring-boot-starter-actuator')
-    compile('org.springframework.boot:spring-boot-starter-web')
-    compile project(':library')
-    testCompile('org.springframework.boot:spring-boot-starter-test')
+	implementation 'org.springframework.boot:spring-boot-starter-actuator'
+	implementation 'org.springframework.boot:spring-boot-starter-web'
+	compile project(':library')
```
Comment (dsyer) [inline complete/application/build.gradle]: I'm not a Gradler, but why is this one still "compile"? isn't that out-moded?

File: initial/application/build.gradle [hunk 2]
```diff
@@ -1,30 +1,19 @@
...
-repositories { mavenCentral() }
 
 dependencies {
-    compile('org.springframework.boot:spring-boot-starter-actuator')
-    compile('org.springframework.boot:spring-boot-starter-web')
-    compile project(':library')
-    testCompile('org.springframework.boot:spring-boot-starter-test')
+	implementation 'org.springframework.boot:spring-boot-starter-actuator'
+	implementation 'org.springframework.boot:spring-boot-starter-web'
+	testImplementation 'org.springframework.boot:spring-boot-starter-test'
```
Comment (dsyer) [inline initial/application/build.gradle:18]: Missing dependency on library? I guess that's OK for the initial?

File: complete/application/build.gradle (hunk 1 above)
Comment (Buzzardo) [reply complete/application/build.gradle]: I couldn't get the implementation way to work, so I went back to the compile way. If someone has another way, I'd love to see it.

Comment (dsyer) [reply complete/application/build.gradle]: Calling @wilkinsona ...

Comment (wilkinsona) [reply complete/application/build.gradle]: It should work with `library` and does so for me:

```
[$]: ./gradlew build
Downloading https://services.gradle.org/distributions/gradle-4.6-bin.zip
......................................................................
Unzipping /Users/awilkinson/.gradle/wrapper/dists/gradle-4.6-bin/4jp4stjndanmxuerzfseyb6wo/gradle-4.6-bin.zip to /Users/awilkinson/.gradle/wrapper/dists/gradle-4.6-bin/4jp4stjndanmxuerzfseyb6wo
Set executable permissions for: /Users/awilkinson/.gradle/wrapper/dists/gradle-4.6-bin/4jp4stjndanmxuerzfseyb6wo/gradle-4.6/bin/gradle
Starting a Gradle Daemon (subsequent builds will be faster)

> Task :application:test 
2019-09-27 09:04:12.806  INFO 14694 --- [       Thread-5] o.s.s.concurrent.ThreadPoolTaskExecutor  : Shutting down ExecutorService 'applicationTaskExecutor'


BUILD SUCCESSFUL in 37s
9 actionable tasks: 9 executed
[$]: git diff HEAD
diff --git a/complete/application/build.gradle b/complete/application/build.gradle
index 28336a1..b452658 100644
--- a/complete/application/build.gradle
+++ b/complete/application/build.gradle
@@ -15,6 +15,6 @@ repositories {
 dependencies {
        implementation 'org.springframework.boot:spring-boot-starter-actuator'
        implementation 'org.springframework.boot:spring-boot-starter-web'
-       compile project(':library')
+       implementation project(':library')
        testImplementation 'org.springframework.boot:spring-boot-starter-test'
 }
```

File: initial/application/build.gradle (hunk 2 above)
Comment (Buzzardo) [reply initial/application/build.gradle:18]: (I thought I answered this, but I don't see it. Sorry if there are two answers.)

I thought I would use only what the Initializr generated in the initial project and have the complete project contain the bits as they look after editing.

**Pull request merged**